### PR TITLE
Improved Indentation and added Comments in the codebase.

### LINKED ISSUE #117: Improving Indentation and adding Comments in codebase
## Problem
There are few indentation issues and the code lacks the comments that would help it be easier to read.

## Description
Adding suitable comments in the codebase and maintaining-proper indentation throughout

## Additional context
Comment serve as a reminder to change something in the future. Code commenting also becomes invaluable when you want to ‘comment out’ code. This is when you turn a block of code into a comment because you don’t want that code to be run, but you still want to keep it on hand in case you want to re-add it.

## Implementation
- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by ro4i7**
## Description

Fixes #117 

Improved Indentation and added Comments in the codebase.

Please give wherever changes need.

## Changes Made
[List the changes you have made in bullet points.]

## Checklist
- [ ] I have read and followed the contributing guidelines.
- [ ] I have tested my changes thoroughly and they work as expected.
- [ ] I have added necessary tests for the changes made.
- [ ] I have updated the documentation to reflect the changes made.
- [ ] My code follows the project's coding style and standards.
- [ ] I have added appropriate commit messages and comments for my changes.

## Related Issues
[If your changes relate to any existing issues, please list them here.]

## Additional Notes
[Provide any additional notes or context that may be helpful for the reviewers.]

Feel free to modify and customize this template as needed to fit the specific needs of the GCP Scanner project.

### CODE REVIEW

**Comment by ro4i7** [comment]
@mshudrak  please check this

**Comment by ro4i7** [comment]
@mshudrak  ,in the review section, there is no showing pylint error for other files except `credsdb.py` is there any check still needed?

**Comment by mshudrak** [comment]
I don't see that tests even run now. Additionally, this CL is too large for review. I'd advice you to split it per file. I also see a lot of comments you added to be redundant. See one I made for example.

File: scanner.py [hunk 1]
```diff
@@ -14,5 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Import the scanner module from the gcp_scanner package
```
Comment (mshudrak) [inline scanner.py:17]: This is redundant

File: scanner.py [hunk 2]
```diff
@@ -14,5 +14,8 @@
...
 from src.gcp_scanner import scanner
-scanner.main()
\ No newline at end of file
+
+# Call the main function of the scanner module to start the scanning process
```
Comment (mshudrak) [inline scanner.py:20]: This is redundant

File: src/gcp_scanner/__main__.py [hunk 3]
```diff
@@ -16,7 +16,10 @@
 
 """
 
+# Importing the scanner module
```
Comment (mshudrak) [inline src/gcp_scanner/__main__.py:19]: redundant

File: src/gcp_scanner/__main__.py [hunk 4]
```diff
@@ -16,7 +16,10 @@
...
 from . import scanner
 
+# Checking if the code is running as the main module
 if __name__ == '__main__':
+  # Calling the main function of the scanner module
```
Comment (mshudrak) [inline src/gcp_scanner/__main__.py:24]: This is redundant

File: src/gcp_scanner/arguments.py [hunk 5]
```diff
@@ -20,6 +20,8 @@
 import argparse
 import logging
 
+
+# Define a function to create an argument parser using the argparse module
```
Comment (mshudrak) [inline src/gcp_scanner/arguments.py:24]: This is redundant

File: src/gcp_scanner/arguments.py [hunk 6]
```diff
@@ -31,102 +33,117 @@ def arg_parser():
     argparse.Namespace: A namespace object containing the parsed command-line
     arguments.
   """
+  # Create a new parser object
```
Comment (mshudrak) [inline src/gcp_scanner/arguments.py:36]: This is redundant. I am stopping on this one. You do not need to comment obvious sections of the code.

**Comment by ro4i7** [comment]
> I don't see that tests even run now. Additionally, this CL is too large for review. I'd advice you to split it per file. I also see a lot of comments you added to be redundant. See one I made for example.

Yeah. I also thought that. Ok, I am working on it

**Comment by ro4i7** [comment]
> > I don't see that tests even run now. Additionally, this CL is too large for review. I'd advice you to split it per file. I also see a lot of comments you added to be redundant. See one I made for example.
> 
> Yeah. I also thought that. Ok, I am working on it

I think, can I close this issue. And start new PR fir group of 3 files. So that it can easy for check?

**Comment by mshudrak** [comment]
Take a look at my other comments, a lot of comments are redundant. We do not need that level of comment details especially for obvious things. Also, we use 2 spaces instead of 4.

**Comment by mshudrak** [comment]
Closing this one due to inactivity. Feel free to split it into small PRs and resubmit.

**Pull request closed without merging**